bin/analysercli

```diff
@@ -0,0 +1,274 @@
+        METRICS = ['cityblock', 'euclidiean', 'l1', 'l2']
+        kwargs = {}
+        try:
+            line = line.split()
+            if len(line) > 0:
+                kwargs['min_samples'] = int(line[0])
+            if len(line) > 1:
+                # Maximum distance between elements.
+                kwargs['eps'] = float(line[1])
+            if len(line) > 2 and line[2] in METRICS:
+                kwargs['metric'] = line[2]
+        except ValueError:
+            self.help_cluster_hosts_dbscan()
+            return
+        self.get_context().clusterize_dbscan(**kwargs)
+
+    def do_display_hosts(self, _):
+        """Print hosts' ObjectId(s) to stdout"""
+        print ','.join(map(str, self.get_context().host_ids))
+
+    def do_remove_anomalies(self, _):
+        """Remove anomalies from hosts"""
+        try:
+            self.push(self.get_context().remove_anomalies())
+        except (RuntimeError, AttributeError):
+            colorize_log('warning', 'Search anomalies first.')
+
+    def do_select_anomalies(self, _):
+        """Select anomalies from hosts"""
+        try:
+            self.push(self.get_context().select_anomalies())
+        except (RuntimeError, AttributeError):
+            colorize_log('warning', 'Search anomalies first.')
+
+    def do_select_clusters(self, line):
+        """Select provided clusters"""
+        try:
+            cluster_labels = [int(idx) for idx in line.split()]
+            if not cluster_labels:
+                raise ValueError()
+        except ValueError:
+            self.help_select_clusters()
+            return
+        try:
+            self.push(self.get_context().select_clusters(cluster_labels))
+        except (RuntimeError, AttributeError):
+            colorize_log('warning', 'Cluster hosts first.')
+
+    def do_remove_clusters(self, line):
+        """Remove provided clusters"""
+        try:
+            cluster_labels = [int(idx) for idx in line.split()]
+            if not cluster_labels:
+                raise ValueError()
+        except ValueError:
+            self.help_select_clusters()
+            return
+        try:
+            self.push(self.get_context().remove_clusters(cluster_labels))
+        except (RuntimeError, AttributeError):
+            colorize_log('warning', 'Cluster hosts first.')
+
+    def do_select_centers(self, _):
+        """Select cluster centers"""
+        try:
+            self.push(self.get_context().select_cluster_centers())
+        except (RuntimeError, AttributeError):
+            colorize_log('warning', 'Cluster hosts first.')
+
+    @staticmethod
+    def help_count():
+        """Help for do_count command"""
+        log = 'count\n'
+        log += '\tPrint host count to sdout'
+        colorize_log('usage', log)
+
+    @staticmethod
+    def help_plot2d():
+        """Print help for do_plot2d command"""
+        log = 'plot2d\n'
+        log += '\tPlot hosts in two dimensions'
```

Comment:
This is bad. You need to avoid `+` on strings (particularly when they're useless). Here write:
```Python
log = """plot2d                                                                 
\tPlot hosts in two dimensions"""
```
or:
```Python
log = "plot2d\n\tPlot hosts in two dimensions"
```
(both are fine for me)